README.md

```diff
@@ -1,4 +1,154 @@
-# `hello-juju-charm`
+# Hello, Juju! Charm
 
-This repository contains the source code for introductory charms
-for new Juju users.
+This [charm](https://charmhub.io/hello-juju) is a demonstration machine charm
+written with the [Charmed Operator Framework](https://github.com/canonical/operator).
+
+It deploys a simple Python [Flask](https://flask.palletsprojects.com/en/2.0.x/)
+[web application](https://github.com/juju/hello-juju). The application itself is
+simple, and counts the number of times the root URL has been requested and stores the
+count in a database. With no database relation defined, this information is stored in
+a sqlite database.
+
+This charm also supports a [relation](https://juju.is/docs/sdk/relations) to the
+[PostgreSQL charm](https://charmhub.io/postgresql). When this relation is established
+the Flask application is automatically configured to use the PostgreSQL database
+to store its count.
+
+## Quickstart
+
+Assuming you already have Juju installed and bootstrapped on a cloud, such as LXD:
+
+```bash
+# Create a model
+$ juju add-model dev
+# Deploy the charm
+$ juju deploy hello-juju
+# Wait for the deployment to complete
+$ juju status
+Model  Controller  Cloud/Region         Version  SLA          Timestamp
+dev    lxd         localhost/localhost  2.9.1    unsupported  14:46:51+01:00
+
+App         Version  Status  Scale  Charm       Store  Channel  Rev  OS      Message
+hello-juju           active      1  hello-juju  local             0  ubuntu
+
+Unit           Workload  Agent  Machine  Public address  Ports   Message
+hello-juju/0*  active    idle   0        10.14.25.117    80/tcp
+
+Machine  State    DNS           Inst id        Series  AZ  Message
+0        started  10.14.25.117  juju-9f46aa-0  focal       Running
+```
+
+You should be able to visit [http://10.14.25.117](http://10.14.25.117)
+in your browser.
+
+Once you've visited/refreshed a couple of times, visit
+[http://10.14.25.117/greetings](http://10.14.25.117/greetings) to see a count
+of how many times the page has been requested.
+
+## Using PostgreSQL
+
+To use PostgreSQL as the backing database, deploy the charm, then deploy PostgreSQL
+and relate the two applications, specifying the `db` interface.
+
+Assuming you already have the `hello-juju` charm deployed, you can start using
+PostgreSQL like so:
+
+```bash
+# Deploy PostgreSQL
+$ juju deploy postgresql
+# Relate the two applications
+$ juju relate hello-juju postgresql:db
+# Check the status:
+$ juju status --relations
```

Comment:
I got an error when trying to repro these steps: https://pastebin.canonical.com/p/wHpXyHfYxp/ -- it looks like the root cause is the PostgreSQL Python client `psyopg2` is not being installed: `ModuleNotFoundError: No module named 'psycopg2'`

Reply:
FWIW the reason why this charm originally specifed `pg8000` rather than `psycopg2` is that it is _much_ easier to install reliably.

Reply:
I should have mentioned that until the other PR is merged, the deploy command should be:

`juju deploy ./hello-juju-charm --config application-repo=https://github.com/jnsgruk/hello-juju`

Saying that @timClicks, you might have a point there. I've switched back to `pg8000`. A minor change in this repo to just update the connection string provided by the postgresql relation. Thanks! :)